### PR TITLE
Add Xojo lexer.

### PR DESCRIPTION
**Pull request by jimmckay**
Started with a fresh branch.
Simplified the lexer and removed unneeded parsing.

### CODE REVIEW

**Comment by pyrmont** [comment]
@jimmckay There's [an error](https://travis-ci.org/rouge-ruby/rouge/jobs/538335878) in the code that's causing Travis to fail. You should be able to run the same tests locally first before submitting. Running `rake spec` and `rubocop` from the root of the project should do the trick.

**Comment by pyrmont** [comment]
Thanks for responding so quickly, @jimmckay. I'm looking through the lexer but have an additional request in the meantime. Could you add a spec? A simple one like the one for Rust would be fine:

https://github.com/rouge-ruby/rouge/blob/e5ec8df0d48fdd31b53c871aa70d8dade481124a/spec/lexers/rust_spec.rb#L1-L22

You can skip the guessing by source test as it's not relevant here.

**Comment by jimmckay** [comment]
Sure. I'll take a look.

**Comment by pyrmont** [comment]
Cheers :) Might not be able to get through a review tonight but will try to have it done by tomorrow!

**Comment by jimmckay** [comment]
Sounds great! Let me know about anything that could be improved. I'm very new to Ruby.

File: lib/rouge/lexers/xojo.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        rule /\b(?:#{keywords.join('|')})\b/i, Keyword
+        rule /\b(?:#{keywords_type.join('|')})\b/i, Keyword::Declaration
+        rule /\b(?:#{operator_words.join('|')})\b/i, Operator
+       
+        rule(
+          %r(<=|>=|<>|[=><\+\-\*\/\\]),
+          Operator
+        )
+
+        rule /[+-]?(\d+\.\d*|\d*\.\d+)(f[+-]?\d+)?/i, Literal::Number::Float
```
Comment (pyrmont) [inline lib/rouge/lexers/xojo.rb]: Two points:

1. This rule will match `5.`. The period should only be part of the number if there is a number immediately following it, right?

2. How does the `f` syntax work? I didn't see that in the visual sample.

File: lib/rouge/lexers/xojo.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+module Rouge
+  module Lexers
+    class Xojo < RegexLexer
+      title "Xojo"
+      desc "Xojo"
+      tag 'xojo'
+      aliases 'realbasic'
+      filenames '*.xojo_code', '*.xojo_window', '*.xojo_toolbar', '*.xojo_menu'
+
+        keywords = %w(
```
[the remark this answers is not in the document]
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Fixed

File: lib/rouge/lexers/xojo.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+          int32 int64 integer ostype pstring ptr short single
+          single string structure variant uinteger uint8 uint16 uint32 uint64
+          ushort windowptr wstring
+        )
+
+        operator_words = %w(
+          addressof and as in is isa mod not or xor
+        )
+
+      state :whitespace do
```
[the remark this answers is not in the document]
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Considering the simplicity of this lexer, I don't think states are really needed at this point. I've moved everything to the root state. Check that this seems correct.

File: lib/rouge/lexers/xojo.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        rule /\s+/, Text
+        rule /rem\b.*?$/i, Comment::Single
```
[the remark this answers is not in the document]
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Yes, it is strange. A throwback to the old Basic language. Rem being short for Remark. I've added an example to the visual sample.

File: lib/rouge/lexers/xojo.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        rule /\/\/.*$/, Comment::Single
+        rule /\#tag Note.*\#tag EndNote/m, Comment::Doc
+        rule /".*"/, Literal::String
```
[the remark this answers is not in the document]
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Xojo escapes " by placing two "'s together with no space between. 
Something like "a ""Quoted"" quote"
I've updated the regex and added a test to the sample.

File: lib/rouge/lexers/xojo.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        rule /\b(?:#{operator_words.join('|')})\b/i, Operator
+       
+        rule(
+          %r(<=|>=|<>|[=><\+\-\*\/\\]),
+          Operator
+        )
+
+        rule /[+-]?(\d+\.\d*|\d*\.\d+)(f[+-]?\d+)?/i, Literal::Number::Float
+        rule /[+-]?\d+/, Literal::Number::Integer
+        rule /&[CH][0-9a-f]+/i, Literal::Number::Integer
```
[the remark this answers is not in the document]
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Yes. Fixed

File: lib/rouge/lexers/xojo.rb (hunk 1 above)
Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: 1. Fixed
2. That is a copy/paste error... fixed in latest commit

Comment (jimmckay) [reply lib/rouge/lexers/xojo.rb]: Sorry, I misread that. It should allow 5. as a valid float value.  .5 is also allowed. Updated lexer and sample in the latest commit.

**Comment by pyrmont** [comment]
Thanks for putting this together, @jimmckay! Much appreciated :)

**Pull request merged**